rfcs/0114-nixos-code-of-conduct.md

```diff
@@ -0,0 +1,108 @@
+inappropriate behavior when interacting with the community. This RFC aims
+to establish the shared values, so they can be used to identify behavior
+which is disruptive to the community.
+
+This RFC does not intend to define moderation practices. This is intended to only lay
+a framework of shared values for the community.
+
+# Motivation
+[motivation]: #motivation
+
+The NixOS community struggles with a clearly defined way in which someone's behavior may
+be disruptive. A well-defined Code of Conduct gives an explicit
+set of expectations for all contributors, which will also make it easier to identify
+when behavior becomes disruptive.
+
+# Detailed design
+[design]: #detailed-design
+
+These statutes are taken from [rust-lang's Code of Conduct](https://www.rust-lang.org/policies/code-of-conduct),
+which provides a great compromise between explicit behaviors and subjective goals.
+Only the moderation body name has been changed.
+
+- We are committed to providing a friendly, safe and welcoming environment for
+all, regardless of level of experience, gender identity and expression,
+sexual orientation, disability, personal appearance, body size, race, ethnicity, age,
+religion, nationality, or other similar characteristics.
+- Please avoid using overtly sexual aliases or other nicknames that might
+detract from a friendly, safe and welcoming environment for all.
+- Please be kind and courteous. There’s no need to be mean or rude.
+- Respect that people have differences of opinion and that every design or
+implementation choice carries a trade-off and numerous costs. There is seldom a right answer.
+- Please keep unstructured critique to a minimum. If you have solid ideas
+you want to experiment with, make a fork and see how it works.
+- We will exclude you from interaction if you insult, demean or harass anyone.
+That is not welcome behavior. We interpret the term “harassment” as including the definition in the
+[Citizen Code of Conduct](https://github.com/stumpsyn/policies/blob/master/citizen_code_of_conduct.md);
+if you have any lack of clarity about what might be included in that concept,
+please read their definition. In particular, we don’t tolerate behavior that excludes
+people in socially marginalized groups.
+- Private harassment is also unacceptable. No matter who you are, if you feel
+you have been or are being harassed or made uncomfortable by a community member,
+please contact one of the channel ops or any of the NixOS moderation team immediately.
+Whether you’re a regular contributor or a newcomer, we care about making this community
+a safe place for you and we’ve got your back.
+- Likewise any spamming, trolling, flaming, baiting or other attention-stealing behavior is not welcome.
+
+# Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+There have been a few spectacular failures of communication and good faith within the community.
```

Comment:
I think the proposed statutes are fine, but I'm not convinced that that they alone would have resolved the three threads referenced here.  If you're going to reference these threads to motivate the statutes then you'd need to elaborate on how a hypothetical moderator would have applied the statutes to improve the outcome of those threads.

As a concrete example, what should a moderator have done differently within the block evasion thread compared to what actually transpired?

Reply:
I guess the _hope_ is that it would take a reasonably short time to reach the conclusion that the proposed  guidelines imply «please don't do it», and there would be something that can imply one way or another in the first place.

Reply:
It might not be useful to use the block evasion thread as an example, because many posts in it were split into a private conversation.

Reply:
Then perhaps we should remove it as an example from this section if it is not an instructive example?

I would still like to see at least one illustration of how one of these motivating examples would have been moderated differently if these guidelines had been place

Reply:
> need to elaborate on how a hypothetical moderator would have applied the statutes to improve the outcome of those threads.

That's specifically out-of-scope for the RFC. I'm just defining the criteria in which people should act, not how it should be enforced. That's far too much scope for a single RFC.

When a moderator does decide to step in, this CoC is meant to be the basis in which actions are considered.

Reply:
Then perhaps remove the examples?  The only reason I suggest this is that the examples suggestion seems to suggest that this RFC is intended to address those conflicts, but it seems like that connection has yet to be made

Reply:
Where does it suggest it? I need to change it if it does.

> This RFC does not intend to define moderation practices. This is intended to only lay a framework of shared values for the community.

Reply:
The RFC says:

> In conjunction with a moderation team (out-of-scope for this RFC), these incidents could have been
better arbitrated as to have a more satisfactory resolution before escalation. This
Code of Conduct will better equip the moderation process by providing clear expectations
for behavior within the community.

… and so I'm asking if this claim could be substantiated by illustrating how it would better equip the moderation process.  If substantiating this claim is out of scope of the RFC then perhaps remove the claim?

Reply:
I added the PR which spawned the discourse discussion.

Reply:
> … and so I'm asking if this claim could be substantiated by illustrating how it would better equip the moderation process. If substantiating this claim is out of scope of the RFC then perhaps remove the claim?

Because the current process is essentially graham getting bat phoned that "something is wrong", after the fact. If there was a moderation body then this CoC would be used to identify what behavior was disruptive.

In the case of [this incident](https://github.com/NixOS/nixpkgs/pull/120729#discussion_r621885348), a community member took it upon themselves to act as an arbitrator (which is an indicator that there is a need for a moderation body which is easy to contact). Having an CoC would at least be a guiding reference where the behaviors of both individuals could be judged in a non-partial manner; instead of using language like "be nicer", the language could have been, "Please be kind and courteous. There’s no need to be mean or rude. It's one of the community's core values".

A more active moderation team (that is also easy to contact a member) is still needed for arbitration or moderation. But that's out-of-scope.

Reply:
Absolutely not. Volunteer interaction is what community is.
Thinking a moderation body serves community interests is as far-fetched as thinking the CoC content is.
Not that it follows that a moderation body would follow it either.